transformer_sidecar/src/transformer_sidecar/watched_directory.py

```diff
@@ -0,0 +1,139 @@
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import logging
+import os
+import time
+from enum import Enum
+from pathlib import Path
+from queue import Queue
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+
+from transformer_sidecar.object_store_uploader import ObjectStoreUploader
+from transformer_sidecar.servicex_adapter import ServiceXAdapter
+
+
+class WatchedDirectory:
+    class TransformStatus(Enum):
+        READY = 0
+        RUNNING = 1
+        SUCCESS = 2
+        FAILURE = 3
+
+    def __init__(self, path: Path, result_upload_queue: Queue,
+                 logger: logging.Logger, servicex: ServiceXAdapter):
+        self.path = path
+        self.logger = logger
+        self.observer = Observer()
+        self.event_handler = TransformerEventHandler(result_upload_queue,
+                                                     logger,
+                                                     servicex,
+                                                     self)
+
+        self.status = WatchedDirectory.TransformStatus.READY
+        self.events = 0
+        self.total_events = 0
+        self.total_size = 0
+
+    def start(self):
+        self.observer.schedule(self.event_handler, str(self.path))
+        self.status = WatchedDirectory.TransformStatus.RUNNING
+        self.observer.start()
+
+    def stop(self, success: bool):
+        self.logger.info("Shutting down watched directory thread")
+        self.observer.stop()
+        self.status = WatchedDirectory.TransformStatus.SUCCESS \
+            if success \
+            else WatchedDirectory.TransformStatus.FAILURE
+
+
+class TransformerEventHandler(FileSystemEventHandler):
+    def __init__(self, result_upload_queue: Queue,
+                 logger: logging.Logger,
+                 servicex: ServiceXAdapter,
+                 my_watched_directory: WatchedDirectory):
+        self.result_upload_queue = result_upload_queue
+        self.servicex = servicex
+        self.logger = logger
+        self.watched_directory = my_watched_directory
+
+    def on_created(self, event):
+        """
+        This event is triggered whenever a new file is created. We assume that the writing
+        of the transformer output file is pretty atomic. We do hang around and wait for
+        the number of bytes to stop changing.
+
+        We also handle .done file which is written to signal that the job is done
+        """
+        # Nothing to do on directory events or log files
+        if event.is_directory or event.src_path.endswith('.log'):
+            return
+
+        # Look for signal that the job is done
+        if event.src_path.endswith(".done"):
+            self.watched_directory.stop(success=True)
+
+            # Terminate the upload thread too
+            self.result_upload_queue.put(ObjectStoreUploader.WorkQueueItem(None))
```

Comment:
Actually the try/catch is useless since in the catch it will try to put something to the queue (which is what caused the exception in the first place) - I deleted that try/catch and just let the exception kill the watched directory